pkg/hubble/container/ring_test.go

```diff
@@ -91,6 +94,76 @@ func BenchmarkTimeLibRingRead(b *testing.B) {
 	})
 }
 
+func BenchmarkRingBufferNoReaders(b *testing.B) {
+	benchmarkRingBuffer(b, benchmarkRingBufferOptions{})
+}
+
+func BenchmarkRingBufferOneReader(b *testing.B) {
+	benchmarkRingBuffer(b, benchmarkRingBufferOptions{
+		readers: 1,
+	})
+}
+
+func BenchmarkRingBufferEightReaders(b *testing.B) {
+	benchmarkRingBuffer(b, benchmarkRingBufferOptions{
+		readers: 8,
+	})
+}
+
+type benchmarkRingBufferOptions struct {
+	capacity capacity
+	readers  int
+}
+
+// benchmarkRingBuffer benchmarks RingBuffer's writes and reads with multiple
+// readers.
+func benchmarkRingBuffer(b *testing.B, options benchmarkRingBufferOptions) {
+	b.ReportAllocs()
+
+	if options.capacity == capacity(0) {
+		options.capacity = benchmarkCapacity
+	}
+
+	r := NewRing(options.capacity)
+
+	ctx, cancel := context.WithCancel(context.Background())
+
+	var readersReady sync.WaitGroup
+	var readersDone sync.WaitGroup
+	for i := 0; i < options.readers; i++ {
+		readersReady.Add(1)
+		readersDone.Add(1)
+		rr := NewRingReader(r, 0)
+		go func(readerIndex int) {
+			readersReady.Done()
+			defer readersDone.Done()
+			for i := 0; i < b.N; i++ {
+				if rr.NextFollow(ctx) == nil {
+					// Ignore dropped events.
+					// b.Logf("reader %d dropped event %d", readerIndex, i)
+				}
```

Comment:
Since we call `cancel` below, do I understand correctly that his test can succeed without any read?